src/Hal/Serializer/CollectionNormalizer.php

```diff
@@ -68,15 +69,17 @@ public function normalize($object, $format = null, array $context = [])
         $resourceClass = $this->resourceClassResolver->getResourceClass($object, $context['resource_class'] ?? null, true);
         $context = $this->initContext($resourceClass, $context);
         $parsed = IriHelper::parseIri($context['request_uri'] ?? '/', $this->pageParameterName);
-        $paginated = $isPaginator = $object instanceof PaginatorInterface;
 
-        $currentPage = $lastPage = $itemsPerPage = null;
```

Comment:
I think those changes introduce the PHPStan errors. It should probably be reverted.

Reply:
Yep, see https://github.com/api-platform/core/pull/1292#discussion_r130375281

Reply:
Feels weird to me to declare empty variables but maybe it's better... Your call @dunglas.

Reply:
It makes the execution path more predictible for static analysis tools (and even for humans IMO).

Reply:
Humans > Robots